examples/mondrian/1-quickstart/plot_main-tutorial-mondrian-regression.py

```diff
@@ -0,0 +1,181 @@
+
+
+X_train_temp, X_test, y_train_temp, y_test = train_test_split(
+    X, y, test_size=0.2, random_state=0
+)
+groups_train_temp, groups_test, _, _ = train_test_split(
+    groups, y, test_size=0.2, random_state=0
+)
+X_cal, X_train, y_cal, y_train = train_test_split(
+    X_train_temp, y_train_temp, test_size=0.5, random_state=0
+)
+groups_cal, groups_train, _, _ = train_test_split(
+    groups_train_temp, y_train_temp, test_size=0.5, random_state=0
+)
+
+
+##############################################################################
+# We plot the training set, the calibration set, and the test set.
+
+
+f, ax = plt.subplots(1, 3, figsize=(15, 5))
+ax[0].scatter(X_train, y_train, c=groups_train)
+ax[0].set_title("Train set")
+ax[1].scatter(X_cal, y_cal, c=groups_cal)
+ax[1].set_title("Calibration set")
+ax[2].scatter(X_test, y_test, c=groups_test)
+ax[2].set_title("Test set")
+plt.show()
+
+
+##############################################################################
+# 3. Fit a random forest regressor on the training set.
+
+
+rf = RandomForestRegressor(n_estimators=100)
+rf.fit(X_train, y_train)
+
+
+##############################################################################
+# 4. Fit a MapieRegressor and a MondrianCP on the calibration set.
+
+
+mapie_regressor = MapieRegressor(rf, cv="prefit")
+mondrian_regressor = MondrianCP(MapieRegressor(rf, cv="prefit"))
+mapie_regressor.fit(X_cal, y_cal)
+mondrian_regressor.fit(X_cal, y_cal, groups=groups_cal)
+
+
+##############################################################################
+# 5. Predict the prediction intervals on the test set with both methods.
+
+
+_, y_pss_split = mapie_regressor.predict(X_test, alpha=.1)
+_, y_pss_mondrian = mondrian_regressor.predict(
+    X_test, groups=groups_test, alpha=.1
+)
+
+
+##############################################################################
+# 6. Compare the coverage by groups, plot both methods side by side.
+
+
+coverages = {}
+for group in np.unique(groups_test):
+    coverages[group] = {}
+    coverages[group]["split"] = regression_coverage_score_v2(
+        y_test[groups_test == group], y_pss_split[groups_test == group]
+    )
+    coverages[group]["mondrian"] = regression_coverage_score_v2(
+        y_test[groups_test == group], y_pss_mondrian[groups_test == group]
+    )
+
+
+# Plot the coverage by groups, plot both methods side by side
+plt.bar(
+    np.arange(len(coverages)) * 2,
+    [float(coverages[group]["split"]) for group in coverages],
+    label="Split"
+)
+plt.bar(
+    np.arange(len(coverages)) * 2 + 1,
+    [float(coverages[group]["mondrian"]) for group in coverages],
+    label="Mondrian"
+)
+plt.xticks(
+    np.arange(len(coverages)) * 2 + .5,
+    [f"Group {group}" for group in coverages],
+    rotation=45
+)
+plt.hlines(0.9, -1, 21, label="90% coverage", color="black", linestyle="--")
+plt.ylabel("Coverage")
+plt.legend(loc='upper left', bbox_to_anchor=(1, 1))
+plt.show()
```

Comment:
You might need to fix fig size before (text is cut off at the moment).

<img width="668" alt="Screenshot 2024-08-26 at 16 33 48" src="https://github.com/user-attachments/assets/fbefaa73-ef09-45a8-95e6-595dba12c61c">

Reply:
done